RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -0,0 +1,113 @@
+#include "RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+using namespace edm;
+
+OnlineBeamSpotESProducer::OnlineBeamSpotESProducer(const edm::ParameterSet& p) {
+  auto cc = setWhatProduced(this);
+
+  transientBS_ = new BeamSpotObjects;
+  theHLTBS_ = new BeamSpotOnlineObjects;
+  theLegacyBS_ = new BeamSpotOnlineObjects;
+  fakeBS_ = new BeamSpotOnlineObjects;
+  fakeBS_->SetBeamWidthX(0.1);
+  fakeBS_->SetBeamWidthY(0.1);
+  fakeBS_->SetSigmaZ(15.);
+  fakeBS_->SetPosition(0., 0., 0.);
+  fakeBS_->SetType(-1);
+
+  bsHLTToken_ = cc.consumesFrom<BeamSpotOnlineObjects, BeamSpotOnlineHLTObjectsRcd>();
+  bsLegacyToken_ = cc.consumesFrom<BeamSpotOnlineObjects, BeamSpotOnlineLegacyObjectsRcd>();
+}
+
+void OnlineBeamSpotESProducer::fillDescription(edm::ParameterSetDescription& desc) {}
+
```

Comment:
fillDescriptions is empty... need at least:
```
  edm::ParameterSetDescription dsc;
  desc.addWithDefaultLabel(dsc);
```
even if there are no parameters...

Reply:
Also wrong syntax, fillDescription argument is ```edm::ConfigurationDescriptions& descriptions```, not ```edm::ParameterSetDescription& desc```: see example:
```
void HLTBool::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
  edm::ParameterSetDescription desc;
  desc.add<bool>("result", false);
  descriptions.add("hltBool", desc);
}
```